From b8653b428ea80478516751d46a4bfe6114c2b096 Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Sat, 10 Sep 2016 20:42:42 +0200
Subject: [PATCH 17/19] gpio-of-helper: pass of_gpio_flags to request

Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>
---
 drivers/gpio/gpio-of-helper.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-of-helper.c b/drivers/gpio/gpio-of-helper.c
index 33415b1..83f362f 100644
--- a/drivers/gpio/gpio-of-helper.c
+++ b/drivers/gpio/gpio-of-helper.c
@@ -57,7 +57,6 @@ struct gpio_of_entry {
 	struct device_node *node;
 	enum gpio_type type;
 	int gpio;
-	enum of_gpio_flags gpio_flags;
 	int irq;
 	const char *name;
 	atomic64_t counter;
@@ -183,6 +182,14 @@ gpio_of_entry_create(struct gpio_of_helper_info *info,
 	}
 	if (of_property_read_bool(node, "dir-changeable"))
 		req_flags |= GPIOF_EXPORT_CHANGEABLE;
+	if (gpio_flags & OF_GPIO_ACTIVE_LOW)
+		req_flags |= GPIOF_ACTIVE_LOW;
+	if (gpio_flags & OF_GPIO_SINGLE_ENDED) {
+		if (gpio_flags & OF_GPIO_ACTIVE_LOW)
+			req_flags |= GPIOF_OPEN_DRAIN;
+		else
+			req_flags |= GPIOF_OPEN_SOURCE;
+	}
 
 	/* request the gpio */
 	err = devm_gpio_request_one(dev, gpio, req_flags, name);
@@ -228,7 +235,6 @@ gpio_of_entry_create(struct gpio_of_helper_info *info,
 	entry->node = of_node_get(node);	/* get node reference */
 	entry->type = type;
 	entry->gpio = gpio;
-	entry->gpio_flags = gpio_flags;
 	entry->irq = irq;
 	entry->name = name;
 
-- 
2.10.1

