From 2489f6836df8bbbc56d5d9083750106fb292a3b1 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 11 Nov 2014 18:19:30 -0600
Subject: [PATCH] trusty: gcc-4.8 (4.8.2-19ubuntu1) has fix

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/kernel/asm-offsets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/kernel/asm-offsets.c b/arch/arm/kernel/asm-offsets.c
index 2d2d608..75fd051 100644
--- a/arch/arm/kernel/asm-offsets.c
+++ b/arch/arm/kernel/asm-offsets.c
@@ -49,7 +49,7 @@
 #error Your compiler is too buggy; it is known to miscompile kernels.
 #error    Known good compilers: 3.3, 4.x
 #endif
-#if GCC_VERSION >= 40800 && GCC_VERSION < 40803
+#if GCC_VERSION >= 40800 && GCC_VERSION < 40802
 #error Your compiler is too buggy; it is known to miscompile kernels
 #error and result in filesystem corruption and oopses.
 #endif
-- 
2.1.1

