From 613ae03cba79cb8bb78df29c8e88401b50363eec Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 11 Nov 2014 18:19:30 -0600
Subject: [PATCH 1/5] trusty: gcc-4.8 (4.8.2-19ubuntu1) has fix

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/kernel/asm-offsets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/kernel/asm-offsets.c b/arch/arm/kernel/asm-offsets.c
index 871b826..cd49cb6 100644
--- a/arch/arm/kernel/asm-offsets.c
+++ b/arch/arm/kernel/asm-offsets.c
@@ -50,7 +50,7 @@
 #error Your compiler is too buggy; it is known to miscompile kernels.
 #error    Known good compilers: 3.3, 4.x
 #endif
-#if GCC_VERSION >= 40800 && GCC_VERSION < 40803
+#if GCC_VERSION >= 40800 && GCC_VERSION < 40802
 #error Your compiler is too buggy; it is known to miscompile kernels
 #error and result in filesystem corruption and oopses.
 #endif
-- 
2.1.4

