From 90e9f836ec9bacd89bf2ce5e6336d035c959d327 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 15 Oct 2014 15:40:26 -0500
Subject: [PATCH] clk: fix extra clk_gpio

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/clk/ti/gpio.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/clk/ti/gpio.c b/drivers/clk/ti/gpio.c
index f4c668e..0377807 100644
--- a/drivers/clk/ti/gpio.c
+++ b/drivers/clk/ti/gpio.c
@@ -17,11 +17,6 @@
 #include <linux/err.h>
 #include <linux/device.h>
 
-struct clk_gpio {
-	struct clk_hw	hw;
-	struct gpio_desc *gpiod;
-};
-
 /**
  * DOC: basic gpio controlled clock which can be enabled and disabled
  *      with gpio output
-- 
2.1.1

