From c9da460ed55e55d3857ed188526379d5b634365c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Mon, 2 Jul 2018 03:00:49 +0200
Subject: [PATCH 17/64] sx1257: Disable SX1255 version check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/sx1257.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/lora/sx1257.c b/drivers/net/lora/sx1257.c
index 9e77350e4804..81c42e22f4a2 100644
--- a/drivers/net/lora/sx1257.c
+++ b/drivers/net/lora/sx1257.c
@@ -38,7 +38,7 @@ static int sx1257_probe(struct spi_device *spi)
 	u8 val;
 	int ret;
 
-	if (true) {
+	if (false) {
 		ret = sx1257_read(spi, 0x07, &val);
 		if (ret) {
 			dev_err(&spi->dev, "version read failed\n");
-- 
2.19.1

