From acf0247218ef1343c889684c4ba7da62beb16842 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Mon, 2 Jul 2018 03:13:14 +0200
Subject: [PATCH 20/64] sx1301: Clean up duplicate set_cs error handling
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/sx1301.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/net/lora/sx1301.c b/drivers/net/lora/sx1301.c
index 53c56a0b3d19..67d41e2298a1 100644
--- a/drivers/net/lora/sx1301.c
+++ b/drivers/net/lora/sx1301.c
@@ -153,16 +153,12 @@ static int sx1301_radio_set_cs(struct spi_controller *ctrl, bool enable)
 
 static void sx1301_radio_spi_set_cs(struct spi_device *spi, bool enable)
 {
-	int ret;
-
 	dev_dbg(&spi->dev, "setting SPI CS to %s\n", enable ? "1" : "0");
 
 	if (enable)
 		return;
 
-	ret = sx1301_radio_set_cs(spi->controller, enable);
-	if (ret)
-		dev_warn(&spi->dev, "failed to write CS (%d)\n", ret);
+	sx1301_radio_set_cs(spi->controller, enable);
 }
 
 static int sx1301_radio_spi_transfer_one(struct spi_controller *ctrl,
-- 
2.19.1

