From e47fc9a76fee144d929c700712686725662c7048 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Sat, 14 Jul 2018 02:24:49 +0200
Subject: [PATCH 30/64] net: lora: nllink: Add module license
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 net/lora/netlink.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/lora/netlink.c b/net/lora/netlink.c
index 065b331e8779..393e9ce7bf91 100644
--- a/net/lora/netlink.c
+++ b/net/lora/netlink.c
@@ -87,3 +87,7 @@ static void __exit nllora_exit(void)
 
 subsys_initcall(nllora_init);
 module_exit(nllora_exit);
+
+MODULE_DESCRIPTION("LoRa netlink driver");
+MODULE_AUTHOR("Andreas Färber <afaerber@suse.de>");
+MODULE_LICENSE("GPL");
-- 
2.19.1

