From b417099f9b3e009a6ba0678d9d62f0234984634d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Sat, 14 Jul 2018 16:05:47 +0200
Subject: [PATCH 31/64] net: lora: ting01m: Prepare GPIO support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/ting01m.c | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/drivers/net/lora/ting01m.c b/drivers/net/lora/ting01m.c
index cf796bcdd35e..13ec98996e56 100644
--- a/drivers/net/lora/ting01m.c
+++ b/drivers/net/lora/ting01m.c
@@ -121,6 +121,27 @@ static int widora_get_version(struct widora_device *widev, char **version, unsig
 	return -EINVAL;
 }
 
+static int widora_set_gpio(struct widora_device *widev, char bank, char pin, bool enabled, unsigned long timeout)
+{
+	char cmd[] = "AT+Pxx=x";
+
+	cmd[4] = bank;
+	cmd[5] = pin;
+	cmd[7] = enabled ? '1' : '0';
+
+	return widora_simple_cmd(widev, cmd, timeout);
+}
+
+static int widora_set_gpio_pb0(struct widora_device *widev, bool enabled, unsigned long timeout)
+{
+	return widora_set_gpio(widev, 'B', '0', enabled, timeout);
+}
+
+static int widora_set_gpio_pd0(struct widora_device *widev, bool enabled, unsigned long timeout)
+{
+	return widora_set_gpio(widev, 'D', '0', enabled, timeout);
+}
+
 static int widora_receive_buf(struct serdev_device *sdev, const u8 *data, size_t count)
 {
 	struct widora_device *widev = serdev_device_get_drvdata(sdev);
@@ -210,6 +231,20 @@ static int widora_probe(struct serdev_device *sdev)
 	dev_info(&sdev->dev, "firmware version: %s\n", sz);
 	kfree(sz);
 
+	ret = widora_set_gpio_pb0(widev, true, HZ);
+	if (ret) {
+		dev_err(&sdev->dev, "Failed to set GPIO PB0 (%d)\n", ret);
+		serdev_device_close(sdev);
+		return ret;
+	}
+
+	ret = widora_set_gpio_pd0(widev, true, HZ);
+	if (ret) {
+		dev_err(&sdev->dev, "Failed to set GPIO PD0 (%d)\n", ret);
+		serdev_device_close(sdev);
+		return ret;
+	}
+
 	dev_info(&sdev->dev, "Done.\n");
 
 	return 0;
-- 
2.19.1

