From f480c3aa3b7e0f7c44fd767b17716b42abb56b18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Sat, 14 Jul 2018 16:13:01 +0200
Subject: [PATCH 32/64] net: lora: ting01m: Add compatible for Ting-01M
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/ting01m.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/lora/ting01m.c b/drivers/net/lora/ting01m.c
index 13ec98996e56..63f2922c79ee 100644
--- a/drivers/net/lora/ting01m.c
+++ b/drivers/net/lora/ting01m.c
@@ -259,6 +259,7 @@ static void widora_remove(struct serdev_device *sdev)
 
 static const struct of_device_id widora_of_match[] = {
 	{ .compatible = "himalaya,himo-01m" },
+	{ .compatible = "widora,ting-01m" },
 	{}
 };
 MODULE_DEVICE_TABLE(of, widora_of_match);
-- 
2.19.1

