From 0f9a65b8cc8271c4a4776cc5a94118597819ab23 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Sun, 2 Sep 2018 02:57:24 +0200
Subject: [PATCH 57/64] net: lora: sx125x: Always print version
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/sx125x.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/lora/sx125x.c b/drivers/net/lora/sx125x.c
index 3476a59ab19e..dc13d1a9e5bd 100644
--- a/drivers/net/lora/sx125x.c
+++ b/drivers/net/lora/sx125x.c
@@ -57,7 +57,7 @@ static int __maybe_unused sx125x_regmap_probe(struct device *dev, struct regmap
 	dev_set_drvdata(dev, priv);
 	priv->regmap = regmap;
 
-	if (false) {
+	if (true) {
 		ret = regmap_read(priv->regmap, SX1255_VERSION, &val);
 		if (ret) {
 			dev_err(dev, "version read failed\n");
-- 
2.19.1

