From dcc1dc63998911620c9da74323a2274bcd9b74e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Tue, 25 Sep 2018 22:00:34 +0200
Subject: [PATCH 63/64] net: Add constant for FLRC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 include/uapi/linux/if_ether.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/linux/if_ether.h b/include/uapi/linux/if_ether.h
index 7d65ef61252f..0b5c30f78261 100644
--- a/include/uapi/linux/if_ether.h
+++ b/include/uapi/linux/if_ether.h
@@ -151,6 +151,7 @@
 #define ETH_P_LORAWAN	0x00FB		/* LoRaWAN			*/
 #define ETH_P_OOK	0x00FC		/* On/Off Keying modulation	*/
 #define ETH_P_FSK	0x00FD		/* Frequency Shift Keying mod.	*/
+#define ETH_P_FLRC	0x00FE		/* Fast Long Range Communication */
 
 /*
  *	This is an Ethernet frame header.
-- 
2.19.1

