From 87a0f707fe23b89d71678c691b8ca9d9b2870d89 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20F=C3=A4rber?= <afaerber@suse.de>
Date: Tue, 25 Sep 2018 22:08:18 +0200
Subject: [PATCH 64/64] net: lora: sx128x: Check for FLRC, too
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 drivers/net/lora/sx128x.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/lora/sx128x.c b/drivers/net/lora/sx128x.c
index 2f78153bac5d..6a5590d0c247 100644
--- a/drivers/net/lora/sx128x.c
+++ b/drivers/net/lora/sx128x.c
@@ -138,7 +138,8 @@ static netdev_tx_t sx128x_loradev_start_xmit(struct sk_buff *skb, struct net_dev
 
 	netdev_dbg(netdev, "%s\n", __func__);
 
-	if (skb->protocol != htons(ETH_P_LORA)) {
+	if (skb->protocol != htons(ETH_P_LORA) &&
+	    skb->protocol != htons(ETH_P_FLRC)) {
 		kfree_skb(skb);
 		netdev->stats.tx_dropped++;
 		return NETDEV_TX_OK;
-- 
2.19.1

