From e934203540f4d495d1915b3cfef4ac797840d6f5 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 9 Apr 2015 11:56:11 -0500
Subject: [PATCH 2/2] etnaviv: wheezy build fix

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/staging/etnaviv/etnaviv_gem.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_gem.c b/drivers/staging/etnaviv/etnaviv_gem.c
index a8290a8..8fcf46e 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.c
+++ b/drivers/staging/etnaviv/etnaviv_gem.c
@@ -726,7 +726,8 @@ struct get_pages_work {
 static struct page **etnaviv_gem_userptr_do_get_pages(
 	struct etnaviv_gem_object *etnaviv_obj, struct mm_struct *mm, struct task_struct *task)
 {
-	int ret, pinned, npages = etnaviv_obj->base.size >> PAGE_SHIFT;
+	int ret = 0;
+	int pinned, npages = etnaviv_obj->base.size >> PAGE_SHIFT;
 	struct page **pvec;
 	uintptr_t ptr;
 
@@ -790,7 +791,8 @@ static int etnaviv_gem_userptr_get_pages(struct etnaviv_gem_object *etnaviv_obj)
 	struct page **pvec = NULL;
 	struct get_pages_work *work;
 	struct mm_struct *mm;
-	int ret, pinned, npages = etnaviv_obj->base.size >> PAGE_SHIFT;
+	int ret = 0;
+	int pinned, npages = etnaviv_obj->base.size >> PAGE_SHIFT;
 
 	if (etnaviv_obj->userptr.work) {
 		if (IS_ERR(etnaviv_obj->userptr.work)) {
-- 
2.1.4

