From a99f7dff5696413ee8dcf8518f3c8d06ca2dd8fc Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Sun, 11 Jan 2015 14:59:31 +0000
Subject: [PATCH 071/178] staging: etnaviv: remove "add child" kernel message

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_drv.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_drv.c b/drivers/staging/etnaviv/etnaviv_drv.c
index c5e606a..aae6221 100644
--- a/drivers/staging/etnaviv/etnaviv_drv.c
+++ b/drivers/staging/etnaviv/etnaviv_drv.c
@@ -618,12 +618,9 @@ static int etnaviv_pdev_probe(struct platform_device *pdev)
 
 		of_platform_populate(node, NULL, NULL, dev);
 
-		for_each_available_child_of_node(node, child_np) {
-			DRM_INFO("add child %s\n", child_np->name);
-
+		for_each_available_child_of_node(node, child_np)
 			component_match_add(dev, &match, etnaviv_compare,
 					    child_np);
-		}
 	} else if (dev->platform_data) {
 		char **names = dev->platform_data;
 		unsigned i;
-- 
2.6.2

