From ab4f8daff4b96c63fb7454f4bb184ec35d99d747 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Wed, 21 Oct 2015 16:57:29 +0100
Subject: [PATCH 131/162] staging: etnaviv: validate gem_new() flags argument

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_drv.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/staging/etnaviv/etnaviv_drv.c b/drivers/staging/etnaviv/etnaviv_drv.c
index 4bdafc5..9aefa60 100644
--- a/drivers/staging/etnaviv/etnaviv_drv.c
+++ b/drivers/staging/etnaviv/etnaviv_drv.c
@@ -373,6 +373,10 @@ static int etnaviv_ioctl_gem_new(struct drm_device *dev, void *data,
 {
 	struct drm_etnaviv_gem_new *args = data;
 
+	if (args->flags & ~(ETNA_BO_CACHED | ETNA_BO_WC | ETNA_BO_UNCACHED |
+			    ETNA_BO_FORCE_MMU))
+		return -EINVAL;
+
 	return etnaviv_gem_new_handle(dev, file, args->size,
 			args->flags, &args->handle);
 }
-- 
2.6.2

