From 433a0271cb4cbd93190b0bcb42f8be69bb3a71f8 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Mon, 30 Nov 2015 11:39:49 +0000
Subject: [PATCH 193/195] staging: etnaviv: add some missing sensitive states

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_cmd_parser.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/staging/etnaviv/etnaviv_cmd_parser.c b/drivers/staging/etnaviv/etnaviv_cmd_parser.c
index c9f3e4c..dcfd565 100644
--- a/drivers/staging/etnaviv/etnaviv_cmd_parser.c
+++ b/drivers/staging/etnaviv/etnaviv_cmd_parser.c
@@ -65,8 +65,14 @@ static const struct {
 	ST(0x1520, 8),
 	ST(0x1608, 1),
 	ST(0x1610, 1),
+	ST(0x1658, 1),
+	ST(0x165c, 1),
+	ST(0x1664, 1),
+	ST(0x1668, 1),
+	ST(0x16a4, 1),
 	ST(0x16c0, 8),
 	ST(0x16e0, 8),
+	ST(0x1740, 8),
 	ST(0x2400, 14 * 16),
 	ST(0x10800, 32 * 16),
 #undef ST
-- 
2.6.2

