From 294bf7cb251ba9d3c0b516862297a988f78b9f44 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 9 Oct 2018 16:13:37 -0500
Subject: [PATCH 9/9] udoo: clk-imx6q.c: set IMX6QDL_CLK_CKO

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/clk/imx/clk-imx6q.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/clk/imx/clk-imx6q.c b/drivers/clk/imx/clk-imx6q.c
index 708e7c5590dd..8b3fd77649a2 100644
--- a/drivers/clk/imx/clk-imx6q.c
+++ b/drivers/clk/imx/clk-imx6q.c
@@ -983,5 +983,10 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 	}
 
 	imx_register_uart_clocks(uart_clks);
+
+	if (of_machine_is_compatible("udoo,imx6dl-udoo") ||
+	  of_machine_is_compatible("udoo,imx6q-udoo"))
+		clk_prepare_enable(clk[IMX6QDL_CLK_CKO]);
+
 }
 CLK_OF_DECLARE(imx6q, "fsl,imx6q-ccm", imx6q_clocks_init);
-- 
2.20.1

