From a7e29ce044419fd8eac66052e18356ab8aea379e Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Wed, 28 Jun 2017 22:37:16 +0200
Subject: [PATCH 5/5] uio_pruss: fix unbalanced pm_runtime calls

Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>
---
 drivers/uio/uio_pruss.c | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/drivers/uio/uio_pruss.c b/drivers/uio/uio_pruss.c
index 64d93f4f4dff..64d93f7fb69c 100644
--- a/drivers/uio/uio_pruss.c
+++ b/drivers/uio/uio_pruss.c
@@ -129,6 +129,8 @@ static void pruss_cleanup(struct device *dev, struct uio_pruss_dev *gdev)
 	clk_disable(gdev->pruss_clk);
 	clk_put(gdev->pruss_clk);
 	kfree(gdev);
+	pm_runtime_put(dev);
+	pm_runtime_disable(dev);
 }
 
 static int pruss_probe(struct platform_device *pdev)
@@ -155,6 +157,17 @@ static int pruss_probe(struct platform_device *pdev)
 		ret = -ENOMEM;
 		goto err_free_gdev;
 	}
+
+	pm_runtime_enable(dev);
+	ret = pm_runtime_get_sync(dev);
+	if (ret < 0) {
+		dev_err(dev, "pm_runtime_get_sync() failed\n");
+		pm_runtime_disable(dev);
+		kfree(gdev->info);
+		kfree(gdev);
+		return ret;
+	}
+
 #ifdef CONFIG_ARCH_DAVINCI_DA850
 	/* Power on PRU in case its not done as part of boot-loader */
 	gdev->pruss_clk = clk_get(dev, "pruss");
@@ -172,17 +185,10 @@ static int pruss_probe(struct platform_device *pdev)
 #endif
 
 	if (dev->of_node) {
-		pm_runtime_enable(dev);
-		ret = pm_runtime_get_sync(dev);
-		if (IS_ERR_VALUE(ret)) {
-			dev_err(dev, "pm_runtime_get_sync() failed\n");
-			return ret;
-		}
-
 		ret = of_address_to_resource(dev->of_node, 0, &res);
 		if (IS_ERR_VALUE(ret)) {
 			dev_err(dev, "failed to parse DT reg\n");
-			return ret;
+			goto out_free;
 		}
 		regs_prussio = &res;
 	} else {
-- 
2.18.0

