diff --git a/3rdparty/packaging/builddeb b/3rdparty/packaging/builddeb
index 18aeeeb4..8e8da7fb 100755
--- a/3rdparty/packaging/builddeb
+++ b/3rdparty/packaging/builddeb
@@ -308,6 +308,7 @@ else
 Package: $packagename
 Suggests: $fwpackagename
 Architecture: any
+Depends: initramfs-tools
 Description: Linux kernel, version $version
  This package contains the Linux kernel, modules and corresponding other
  files, version: $version.
