diff --git a/3rdparty/packaging/builddeb b/3rdparty/packaging/builddeb
index 88dbf23..6f9b59d 100755
--- a/3rdparty/packaging/builddeb
+++ b/3rdparty/packaging/builddeb
@@ -45,7 +45,7 @@ create_package() {
 	arm64)
 		debarch=arm64 ;;
 	arm*)
-		debarch=arm$(grep -q CONFIG_AEABI=y $KCONFIG_CONFIG && echo el || true) ;;
+		debarch=arm$(grep -q CONFIG_AEABI=y $KCONFIG_CONFIG && echo hf || true) ;;
 	*)
 		echo "" >&2
 		echo "** ** **  WARNING  ** ** **" >&2
