diff --git a/3rdparty/packaging/builddeb b/3rdparty/packaging/builddeb
index 361f62b9..d612c4eb 100755
--- a/3rdparty/packaging/builddeb
+++ b/3rdparty/packaging/builddeb
@@ -313,6 +313,7 @@ else
 
 Package: $packagename
 Architecture: any
+Depends: initramfs-tools
 Description: Linux kernel, version $version
  This package contains the Linux kernel, modules and corresponding other
  files, version: $version.
