diff --git a/3rdparty/packaging/builddeb b/3rdparty/packaging/builddeb
index b03dd56a..96321fe4 100755
--- a/3rdparty/packaging/builddeb
+++ b/3rdparty/packaging/builddeb
@@ -99,6 +99,7 @@ if is_enabled CONFIG_OF_EARLY_FLATTREE; then
 	# Only some architectures with OF support have this target
 	if [ -d "${srctree}/arch/$SRCARCH/boot/dts" ]; then
 		$MAKE -f $srctree/Makefile INSTALL_DTBS_PATH="$tmpdir/usr/lib/$packagename" dtbs_install
+		$MAKE -f $srctree/Makefile INSTALL_DTBS_PATH="$tmpdir/boot/dtbs/$version" dtbs_install
 	fi
 fi
 
