diff --git a/3rdparty/packaging/builddeb b/3rdparty/packaging/builddeb
index 0cd46e1..97ec11d 100755
--- a/3rdparty/packaging/builddeb
+++ b/3rdparty/packaging/builddeb
@@ -52,7 +52,7 @@ set_debarch() {
 	arm64)
 		debarch=arm64 ;;
 	arm*)
-		debarch=arm$(grep -q CONFIG_AEABI=y $KCONFIG_CONFIG && echo el || true) ;;
+		debarch=arm$(grep -q CONFIG_AEABI=y $KCONFIG_CONFIG && echo hf || true) ;;
 	*)
 		debarch=$(dpkg --print-architecture)
 		echo "" >&2
