From b7bfa014c316b98c5b89d9a55860cbfee0bbcf5c Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
Date: Fri, 21 Feb 2014 18:49:29 +0200
Subject: [PATCH 02/40] gitignore: Ignore DTB files

Signed-off-by: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
---
 .gitignore | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitignore b/.gitignore
index c2ed4ec..0544d50 100644
--- a/.gitignore
+++ b/.gitignore
@@ -114,3 +114,7 @@ all.config
 
 # Kdevelop4
 *.kdev4
+
+# dtb objects
+*.dtb
+*.dtbo
-- 
2.8.1

