From 3eecb919e747fd591c335e689d4cbfcf14d8707a Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
Date: Fri, 21 Feb 2014 18:49:29 +0200
Subject: [PATCH 13/46] gitignore: Ignore DTB files

Signed-off-by: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
---
 .gitignore | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitignore b/.gitignore
index 4ad4a98..5bc7062 100644
--- a/.gitignore
+++ b/.gitignore
@@ -106,3 +106,7 @@ all.config
 
 # Kdevelop4
 *.kdev4
+
+# dtb objects
+*.dtb
+*.dtbo
-- 
2.1.4

