From 288296b3cc85084a3cebd90ab6ee1c0f8c57a7f6 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
Date: Wed, 14 Oct 2015 13:22:38 +0300
Subject: [PATCH 32/38] of: remove bogus return in of_core_init

Signed-off-by: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
---
 drivers/of/base.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/of/base.c b/drivers/of/base.c
index c025728..9bbca56 100644
--- a/drivers/of/base.c
+++ b/drivers/of/base.c
@@ -213,8 +213,6 @@ void __init of_core_init(void)
 	ret = of_overlay_init();
 	if (ret != 0)
 		pr_warn("of_init: of_overlay_init failed!\n");
-
-	return 0;
 }
 
 static struct property *__of_find_property(const struct device_node *np,
-- 
2.8.0.rc3

