From 620cc80cd5ed51a92868178a53c0b6ff927e670d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 5 Aug 2013 08:46:11 -0500
Subject: [PATCH 3/3] saucy: disable stack protector

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4b3b08a..2fed5b3 100644
--- a/Makefile
+++ b/Makefile
@@ -374,7 +374,8 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
 		   -fno-delete-null-pointer-checks \
-		   -Wno-pointer-sign
+		   -Wno-pointer-sign \
+		   -fno-stack-protector
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
-- 
1.7.10.4

