From 6342f4e0e3bcaf35f2da21651c9534e6ff8dfd6d Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 21 Apr 2014 10:06:38 -0500
Subject: [PATCH 2/2] capes: Makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 35c146f..05d4302 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -280,7 +280,13 @@ dtb-$(CONFIG_ARCH_OMAP2PLUS) += omap2420-h4.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-bone.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO4.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO4.dtb \
 	am335x-nano.dtb \
 	am335x-base0033.dtb \
 	am3517-craneboard.dtb \
-- 
1.9.2

