From 94e1c6482fa522ec06492815ffd882de3465b7e8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Sat, 24 Nov 2018 14:16:50 -0600
Subject: [PATCH 18/18] fix: soc-ti-pruss

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/linux/pruss.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/pruss.h b/include/linux/pruss.h
index c797fb1f0c05..251af5d83785 100644
--- a/include/linux/pruss.h
+++ b/include/linux/pruss.h
@@ -134,6 +134,7 @@ struct pruss_intc_config {
 	s8 ch_to_host[MAX_PRU_CHANNELS];
 };
 
+struct rproc;
 struct pruss;
 
 #if IS_ENABLED(CONFIG_TI_PRUSS)
-- 
2.19.1

