From c3a61c00110b847d39acf1d65f2d2b8646e41496 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 30 Mar 2011 08:51:46 -0500
Subject: [PATCH] OMAP3 SGX TI 4.03.00.02 use omap3630 as TI_PLATFORM

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/staging/omap3-sgx/Kbuild |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/omap3-sgx/Kbuild b/drivers/staging/omap3-sgx/Kbuild
index aa483db..dff638e 100644
--- a/drivers/staging/omap3-sgx/Kbuild
+++ b/drivers/staging/omap3-sgx/Kbuild
@@ -7,7 +7,7 @@ export KERNEL_SRC=$(KERNELDIR)
 export KDIR=$(KERNELDIR)
 BUILD = release
 
-TI_PLATFORM = omap3
+TI_PLATFORM = omap3630
 
 export KBUILD_EXTRA_SYMBOLS = `pwd`/services4/srvkm/env/linux/kbuild/Module.symvers
 
-- 
1.7.1

