From e00ed3f819d113a1d9c40e15abde93e2f1cacc9c Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 15 Feb 2011 12:36:41 -0600
Subject: [PATCH] OMAP3 SGX TI 3.01.00.06 use omap3630

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/staging/omap3-sgx/Kbuild |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/omap3-sgx/Kbuild b/drivers/staging/omap3-sgx/Kbuild
index 81fd965..1ee4fde 100644
--- a/drivers/staging/omap3-sgx/Kbuild
+++ b/drivers/staging/omap3-sgx/Kbuild
@@ -4,7 +4,7 @@
 
 BUILD = release
 
-TI_PLATFORM = omap3430
+TI_PLATFORM = omap3630
 
 SGXCORE = 530
 ifeq ($(TI_PLATFORM),ti8168)
-- 
1.7.1

