From 12bd96ef0c89916d5503dc55307da5ec9e832f80 Mon Sep 17 00:00:00 2001
From: Bas van den Berg <b.van.den.berg.nl@gmail.com>
Date: Wed, 4 Dec 2013 14:05:40 +0100
Subject: [PATCH 3/3] OpenGL: fixed IRQ offset

---
 arch/arm/include/asm/irq.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/asm/irq.h b/arch/arm/include/asm/irq.h
index 35c21c3..133056e 100644
--- a/arch/arm/include/asm/irq.h
+++ b/arch/arm/include/asm/irq.h
@@ -1,7 +1,7 @@
 #ifndef __ASM_ARM_IRQ_H
 #define __ASM_ARM_IRQ_H
 
-#define NR_IRQS_LEGACY	16
+#define NR_IRQS_LEGACY	0
 
 #ifndef CONFIG_SPARSE_IRQ
 #include <mach/irqs.h>
-- 
1.8.5.3

