From 55b985240e5e210184c7a7d216475eeef2c417b9 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 16 Sep 2010 19:37:21 -0500
Subject: [PATCH] chacha tablet fix id

---
 drivers/input/touchscreen/chacha.c  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/input/touchscreen/chacha.c b/drivers/input/touchscreen/chacha.c
index 4fc4b6f..2335f7c 100644
--- a/drivers/input/touchscreen/chacha.c
+++ b/drivers/input/touchscreen/chacha.c
@@ -460,7 +460,7 @@ static const struct i2c_device_id chacha_id[] = {
 	{ "chacha", 0 },
 	{ }
 };
-MODULE_DEVICE_TABLE(i2c, chacha);
+MODULE_DEVICE_TABLE(i2c, chacha_id);
 
 static struct i2c_driver chacha_driver = {
 	.driver = {
-- 
1.7.1

