From 679ea26b5ff4b4c58540cc35573c0504ed654d93 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 16 Sep 2010 19:30:27 -0500
Subject: [PATCH] chacha tablet kzalloc fix

---
 drivers/input/touchscreen/chacha.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/input/touchscreen/chacha.c b/drivers/input/touchscreen/chacha.c
index cf9d168..4fc4b6f 100644
--- a/drivers/input/touchscreen/chacha.c
+++ b/drivers/input/touchscreen/chacha.c
@@ -26,6 +26,7 @@
 #include <linux/input.h>
 #include <linux/interrupt.h>
 #include <linux/delay.h>
+#include <linux/slab.h>
 #include <asm/io.h>
 #include <asm/gpio.h>
 #include <linux/i2c.h>
-- 
1.7.1

