--- a/arch/arm/mach-omap2/board-omap3beagle.c	2009-09-09 15:13:59.000000000 -0700
+++ b/arch/arm/mach-omap2/board-omap3beagle.c	2010-03-31 18:45:25.000000000 -0700
@@ -47,6 +47,53 @@
 
 #include "mmc-twl4030.h"
 
+#if defined(CONFIG_KS8851) || defined(CONFIG_KS8851_MODULE)
+#include <linux/spi/spi.h>
+#include <mach/mcspi.h>
+#include <linux/interrupt.h>
+#include <linux/ks8851.h>
+
+#define KSZ8851_SNL_INT	157
+
+static struct omap2_mcspi_device_config ksz8851_mcspi_config = {
+	.turbo_mode	= 0,
+	.single_channel	= 1,  /* 0: slave, 1: master */
+};
+
+static struct ks8851_pdata ksz8851_snl_data = {
+	.irq_flags	= IRQF_TRIGGER_FALLING,
+};
+
+static struct spi_board_info ksz8851_snl_info[] __initdata = {
+	{
+		.modalias	= "ks8851",
+		.bus_num	= 4,
+		.chip_select	= 0,
+		.max_speed_hz	= 36000000,
+		.controller_data = &ksz8851_mcspi_config,
+		.irq		= OMAP_GPIO_IRQ(KSZ8851_SNL_INT),
+		.platform_data	= &ksz8851_snl_data,
+	},
+};
+
+static int __init omap3430_spi_init(void)
+{
+	int irq = irq_to_gpio(ksz8851_snl_info[0].irq);
+
+	if (gpio_request(irq, "spi4_intr") < 0) {
+		printk(KERN_ERR "Failed to request GPIO %d for "
+			"SPI4 interrupt\n", irq);
+		return 1;
+	}
+	gpio_direction_input(irq);
+	spi_register_board_info(ksz8851_snl_info,
+		ARRAY_SIZE(ksz8851_snl_info));
+	return 0;
+}
+#else
+static inline void omap3430_spi_init(void) {}
+#endif
+
 #define GPMC_CS0_BASE  0x60
 #define GPMC_CS_SIZE   0x30
 
@@ -409,6 +456,7 @@ static void __init omap3_beagle_init(voi
 
 	usb_musb_init();
 	omap3beagle_flash_init();
+	omap3430_spi_init();
 
 	/* Ensure SDRC pins are mux'd for self-refresh */
 	omap_cfg_reg(H16_34XX_SDRC_CKE0);
