--- a/drivers/net/Kconfig	2009-03-23 16:12:14.000000000 -0700
+++ b/drivers/net/Kconfig	2010-04-01 10:58:00.000000000 -0700
@@ -1703,6 +1703,15 @@ config TLAN
 
 	  Please email feedback to <torben.mathiasen@compaq.com>.
 
+config KS8851
+	tristate "Micrel KS8851 SPI"
+	depends on SPI
+	select MII
+	select CRC32
+	select EEPROM_93CX6
+	help
+	  SPI driver for Micrel KS8851 SPI attached network chip.
+
 config VIA_RHINE
 	tristate "VIA Rhine support"
 	depends on NET_PCI && PCI
--- a/drivers/net/Makefile	2009-03-23 16:12:14.000000000 -0700
+++ b/drivers/net/Makefile	2010-04-01 10:58:44.000000000 -0700
@@ -82,6 +82,7 @@ obj-$(CONFIG_TC35815) += tc35815.o
 obj-$(CONFIG_SKGE) += skge.o
 obj-$(CONFIG_SKY2) += sky2.o
 obj-$(CONFIG_SKFP) += skfp/
+obj-$(CONFIG_KS8851)	+= ks8851.o
 obj-$(CONFIG_VIA_RHINE) += via-rhine.o
 obj-$(CONFIG_VIA_VELOCITY) += via-velocity.o
 obj-$(CONFIG_ADAPTEC_STARFIRE) += starfire.o
