From bbdaa6f1936aa50dd6fea9f65c0df932204a976f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 13 Mar 2014 14:48:36 -0500
Subject: [PATCH 4/4] build: capes-one-layer

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index a917a9f..d40d24e 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -204,6 +204,7 @@ dtb-$(CONFIG_ARCH_OMAP2PLUS) += omap2420-h4.dtb \
 	am335x-boneblack-ttyO1.dtb \
 	am335x-boneblack-ttyO2.dtb \
 	am335x-boneblack-ttyO4.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
 	am335x-nano.dtb \
 	am335x-base0033.dtb \
 	am3517-evm.dtb \
-- 
1.9.0

