From f29b6e8f7a93875595eea16a109521160198edfa Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 30 Oct 2015 11:14:50 -0500
Subject: [PATCH] ARM: samples seccomp no -m32

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 samples/seccomp/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/samples/seccomp/Makefile b/samples/seccomp/Makefile
index ba942e3ead89..8e4e941dfc18 100644
--- a/samples/seccomp/Makefile
+++ b/samples/seccomp/Makefile
@@ -18,6 +18,7 @@ bpf-direct-objs := bpf-direct.o
 
 # Try to match the kernel target.
 ifndef CONFIG_64BIT
+ifndef CONFIG_ARM
 
 # s390 has -m31 flag to build 31 bit binaries
 ifndef CONFIG_S390
@@ -36,3 +37,4 @@ HOSTLOADLIBES_dropper += $(MFLAG)
 endif
 always := $(hostprogs-m)
 endif
+endif
-- 
2.16.1

