From 00f42ae123fe063194429e56754ba6f7f5de031b Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 25 Feb 2014 10:59:28 -0600
Subject: [PATCH 3/3] build: capes-one-layer

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index d57c1a6..b0d54c1 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -198,6 +198,10 @@ dtb-$(CONFIG_ARCH_OMAP2PLUS) += omap2420-h4.dtb \
 	am335x-evmsk.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-default.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO4.dtb \
 	am335x-nano.dtb \
 	am335x-base0033.dtb \
 	am3517-evm.dtb \
-- 
1.8.5.3

