From 1ef85f0af49b6d03c72f2f3544abdfd26130b1dd Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 5 Jun 2015 10:30:27 -0500
Subject: [PATCH 2/3] tieqep.c: devres: remove devm_request_and_ioremap()

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/misc/tieqep.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/tieqep.c b/drivers/misc/tieqep.c
index e0ae946..d4698c2 100644
--- a/drivers/misc/tieqep.c
+++ b/drivers/misc/tieqep.c
@@ -504,9 +504,9 @@ static int eqep_probe(struct platform_device *pdev)
     }
 
     // Remap the eQEP controller memory into our own memory space
-    eqep->mmio_base = devm_request_and_ioremap(&pdev->dev, r);
-    if (!eqep->mmio_base)
-        return  -EADDRNOTAVAIL;
+    eqep->mmio_base = devm_ioremap_resource(&pdev->dev, r);
+    if (IS_ERR(eqep->mmio_base))
+        return PTR_ERR(eqep->mmio_base);
 
     // Store the platform device in our eQEP data structure for later usage
     eqep->pdev = pdev;
-- 
2.7.0.rc3

