From d3a7bf49545a141c8e9ff519af92ec3080325bea Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 11 Dec 2017 16:44:36 -0600
Subject: [PATCH] btrfs: raid6: pick either raid6_neonx2 or raid6_intx4, saves
 1 second on bootup

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 lib/raid6/algos.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/raid6/algos.c b/lib/raid6/algos.c
index 476994723258..60378032a42b 100644
--- a/lib/raid6/algos.c
+++ b/lib/raid6/algos.c
@@ -81,15 +81,25 @@ const struct raid6_calls * const raid6_algos[] = {
 #if defined(CONFIG_S390)
 	&raid6_s390vx8,
 #endif
+#ifndef CONFIG_KERNEL_MODE_NEON
+	#ifndef CONFIG_SOC_AM33XX
 	&raid6_intx1,
 	&raid6_intx2,
+	#endif
 	&raid6_intx4,
+	#ifndef CONFIG_SOC_AM33XX
 	&raid6_intx8,
+	#endif
+#endif
 #ifdef CONFIG_KERNEL_MODE_NEON
+	#ifndef CONFIG_SOC_AM33XX
 	&raid6_neonx1,
+	#endif
 	&raid6_neonx2,
+	#ifndef CONFIG_SOC_AM33XX
 	&raid6_neonx4,
 	&raid6_neonx8,
+	#endif
 #endif
 	NULL
 };
-- 
2.15.1

