From 799fbcaf7bc1945247e0d3a0921381e4204cf375 Mon Sep 17 00:00:00 2001
From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Tue, 25 Jun 2013 15:51:48 +0200
Subject: [PATCH 022/161] ARM: dts: i.MX27: Add iim node

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
Signed-off-by: Shawn Guo <shawn.guo@linaro.org>
---
 arch/arm/boot/dts/imx27.dtsi |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/imx27.dtsi b/arch/arm/boot/dts/imx27.dtsi
index df425a7..db139bf 100644
--- a/arch/arm/boot/dts/imx27.dtsi
+++ b/arch/arm/boot/dts/imx27.dtsi
@@ -348,6 +348,13 @@
 				clocks = <&clks 41>, <&clks 61>;
 				clock-names = "ipg", "per";
 			};
+
+			iim: iim@10028000 {
+				compatible = "fsl,imx27-iim";
+				reg = <0x10028000 0x1000>;
+				interrupts = <62>;
+				clocks = <&clks 38>;
+			};
 		};
 
 		aipi@10020000 { /* AIPI2 */
-- 
1.7.10.4

