From ae79b546b961f413c97b584e94bc50c7d2223c97 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 14 Jul 2017 11:20:54 -0500
Subject: [PATCH 7/7] adihdmi_drv fix drm_of.h include

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/gpu/drm/i2c/adihdmi_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i2c/adihdmi_drv.c b/drivers/gpu/drm/i2c/adihdmi_drv.c
index 67922246395e..6ab4682be4b9 100644
--- a/drivers/gpu/drm/i2c/adihdmi_drv.c
+++ b/drivers/gpu/drm/i2c/adihdmi_drv.c
@@ -19,7 +19,7 @@
 #include <drm/drm_crtc_helper.h>
 #include <drm/drm_edid.h>
 #include <drm/drm_encoder_slave.h>
-#include <drm_of.h>
+#include <drm/drm_of.h>
 
 #include "adihdmi.h"
 
-- 
2.13.2

