From e87584e251f0aab5eab415eaf521b1fcc9de7c90 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 27 Mar 2012 20:56:03 -0500
Subject: [PATCH] mfd: cpu_is_omap3430 wasnt defined

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mfd/omap-usb-host.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/mfd/omap-usb-host.c b/drivers/mfd/omap-usb-host.c
index 9927129..95e5d7d 100644
--- a/drivers/mfd/omap-usb-host.c
+++ b/drivers/mfd/omap-usb-host.c
@@ -26,6 +26,7 @@
 #include <linux/dma-mapping.h>
 #include <linux/spinlock.h>
 #include <plat/usb.h>
+#include <plat/cpu.h>
 #include <linux/pm_runtime.h>
 
 #define USBHS_DRIVER_NAME	"usbhs_omap"
-- 
1.7.9.1

