From d53f988fc10fe22ec7e64457eac22f264bb72491 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 13 Jan 2011 11:37:56 -0600
Subject: [PATCH] xM audio fix from Ashok

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 sound/soc/omap/omap-mcbsp.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sound/soc/omap/omap-mcbsp.c b/sound/soc/omap/omap-mcbsp.c
index d203f4d..6311e8a 100644
--- a/sound/soc/omap/omap-mcbsp.c
+++ b/sound/soc/omap/omap-mcbsp.c
@@ -358,6 +358,9 @@ static int omap_mcbsp_dai_hw_params(struct snd_pcm_substream *substream,
 	} else if (cpu_is_omap343x()) {
 		dma = omap24xx_dma_reqs[bus_id][substream->stream];
 		port = omap34xx_mcbsp_port[bus_id][substream->stream];
+	} else if (cpu_is_omap3630()) {
+		dma = omap24xx_dma_reqs[bus_id][substream->stream];
+		port = omap34xx_mcbsp_port[bus_id][substream->stream];
 	 } else if (cpu_is_omap44xx()) {
 		dma = omap44xx_dma_reqs[bus_id][substream->stream];
 		port = omap44xx_mcbsp_port[bus_id][substream->stream];
-- 
1.7.1

