From 03925e9976bc0813eeb0505822eee41a1e9a6cee Mon Sep 17 00:00:00 2001
From: Elias Bakken <elias@iagent.no>
Date: Fri, 10 Apr 2015 18:44:56 +0200
Subject: [PATCH 2/2] Cleaned up error reporting

---
 drivers/uio/uio_pruss.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/uio/uio_pruss.c b/drivers/uio/uio_pruss.c
index e235e45..c237040 100644
--- a/drivers/uio/uio_pruss.c
+++ b/drivers/uio/uio_pruss.c
@@ -216,7 +216,6 @@ static int pruss_probe(struct platform_device *pdev)
 				continue;
 			}
 
-			dev_err(&pdev->dev, "Child has %u gpios\n", count);
 			for(cnt=0; cnt<count; cnt++){
 				ret = of_property_read_string_index(child,
 					"pin-names", cnt, &pin_name);
@@ -224,6 +223,10 @@ static int pruss_probe(struct platform_device *pdev)
 					dev_err(&pdev->dev, "Error on pin-name #%d\n", cnt);
 				gpio = of_get_gpio_flags(child, cnt, &flags);
 				ret = devm_gpio_request_one(&pdev->dev, gpio, flags, pin_name);
+				if (ret < 0) {
+		                        dev_err(dev, "Failed to request GPIO %d (%s) flags: '%d', error %d\n",
+					gpio, pin_name, flags, ret);
+		                }
 			}
 		}
 	}
-- 
2.1.4

