From 1fb1add017a442ee8b23aff020dba49daf6e9e34 Mon Sep 17 00:00:00 2001
From: Vishwa Sripathy <vishwanath.bs@ti.com>
Date: Thu, 17 Feb 2011 18:21:41 +0530
Subject: [PATCH 30/48] omap2+ dvfs: fix up SR enable/disable

Signed-off-by: Vishwa Sripathy <vishwanath.bs@ti.com>
---
 arch/arm/mach-omap2/dvfs.c |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/dvfs.c b/arch/arm/mach-omap2/dvfs.c
index 1e5492c..526e0af 100644
--- a/arch/arm/mach-omap2/dvfs.c
+++ b/arch/arm/mach-omap2/dvfs.c
@@ -512,6 +512,7 @@ static int omap_dvfs_voltage_scale(struct omap_vdd_dvfs_info *dvfs_info)
 	struct voltagedomain *voltdm;
 	unsigned long volt;
 	struct omap_vdd_info *vdd;
+	bool is_sr_disabled = false;
 
 	if (!dvfs_info || IS_ERR(dvfs_info)) {
 		pr_warning("%s: VDD specified does not exist!\n", __func__);
@@ -530,7 +531,10 @@ static int omap_dvfs_voltage_scale(struct omap_vdd_dvfs_info *dvfs_info)
 	curr_volt = omap_voltage_get_nom_volt(voltdm);
 
 	/* Disable smartreflex module across voltage and frequency scaling */
-	omap_sr_disable(voltdm);
+	if (curr_volt != volt) {
+		omap_sr_disable(voltdm);
+		is_sr_disabled = true;
+	}
 
 	if (curr_volt == volt) {
 		is_volt_scaled = 1;
@@ -539,7 +543,8 @@ static int omap_dvfs_voltage_scale(struct omap_vdd_dvfs_info *dvfs_info)
 		if (ret) {
 			pr_warning("%s: Unable to scale the %s to %ld volt\n",
 						__func__, voltdm->name, volt);
-			omap_sr_enable(voltdm);
+			if (is_sr_disabled)
+				omap_sr_enable(voltdm);
 			mutex_unlock(&dvfs_info->scaling_mutex);
 			return ret;
 		}
@@ -575,7 +580,8 @@ static int omap_dvfs_voltage_scale(struct omap_vdd_dvfs_info *dvfs_info)
 		omap_voltage_scale_vdd(voltdm, volt);
 
 	/* Enable Smartreflex module */
-	omap_sr_enable(voltdm);
+	if (is_sr_disabled)
+		omap_sr_enable(voltdm);
 
 	mutex_unlock(&dvfs_info->scaling_mutex);
 
-- 
1.7.1

