From 810461a4704c9ee7f7c6c7cc3f4e38e70f29c058 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 11 Nov 2014 11:28:45 -0600
Subject: [PATCH] auto generated: capes: add dtbs to makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 38 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 7c2c7e6..d976e55 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -329,6 +329,44 @@ dtb-$(CONFIG_ARCH_OMAP3) += am3517-craneboard.dtb \
 dtb-$(CONFIG_SOC_AM33XX) += am335x-base0033.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-ttyO5.dtb \
+	am335x-boneblack-ttyO4.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-rtc-01-00a1.dtb \
+	am335x-boneblack-lcd7-01-00a3.dtb \
+	am335x-boneblack-lcd7-01-00a2.dtb \
+	am335x-boneblack-lcd4-01-00a1.dtb \
+	am335x-boneblack-lcd3-01-00a2.dtb \
+	am335x-boneblack-bbb-exp-c.dtb \
+	am335x-boneblack-4dcape-70t.dtb \
+	am335x-boneblack-4dcape-70.dtb \
+	am335x-boneblack-4dcape-43t.dtb \
+	am335x-boneblack-4dcape-43.dtb \
+	am335x-boneblack-crypto-00a0.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
+	am335x-boneblack-bb-view-43.dtb \
+	am335x-boneblack-audio-revb.dtb \
+	am335x-boneblack-audio-reva.dtb \
+	am335x-bone-ttyO5.dtb \
+	am335x-bone-ttyO4.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-rtc-01-00a1.dtb \
+	am335x-bone-lcd7-01-00a3.dtb \
+	am335x-bone-lcd7-01-00a2.dtb \
+	am335x-bone-lcd4-01-00a1.dtb \
+	am335x-bone-lcd3-01-00a2.dtb \
+	am335x-bone-bbb-exp-c.dtb \
+	am335x-bone-4dcape-70t.dtb \
+	am335x-bone-4dcape-70.dtb \
+	am335x-bone-4dcape-43t.dtb \
+	am335x-bone-4dcape-43.dtb \
+	am335x-bone-crypto-00a0.dtb \
+	am335x-bone-cape-bone-argus.dtb \
+	am335x-bone-bb-view-43.dtb \
+	am335x-bone-audio-revb.dtb \
+	am335x-bone-audio-reva.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-nano.dtb \
-- 
2.1.1

