From 0d92393c1ec5d00b2d3d025a2236493fc7fcc66d Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
Date: Fri, 21 Feb 2014 18:49:29 +0200
Subject: [PATCH 01/38] gitignore: Ignore DTB files

Signed-off-by: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
---
 .gitignore | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitignore b/.gitignore
index 0c39aa20b6ba..bdcc9d8bdd1f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -115,3 +115,7 @@ all.config
 
 # Kdevelop4
 *.kdev4
+
+# dtb objects
+*.dtb
+*.dtbo
-- 
2.13.2

