From 4a70068bab6b938e5c4508ee3dda59bbd9616d05 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 12 Sep 2017 11:09:52 -0500
Subject: [PATCH 39/39] bone_capemgr: kill with uboot flag

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/misc/bone_capemgr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/misc/bone_capemgr.c b/drivers/misc/bone_capemgr.c
index f57df896804f..0262a46f2a91 100644
--- a/drivers/misc/bone_capemgr.c
+++ b/drivers/misc/bone_capemgr.c
@@ -1616,6 +1616,9 @@ capemgr_probe(struct platform_device *pdev)
 	long val;
 	char *wbuf, *s, *p, *e;
 
+	if (uboot_capemgr_enabled)
+		return 0;
+
 	/* we don't use platform_data at all; we require OF */
 	if (pnode == NULL)
 		return -ENOTSUPP;
-- 
2.14.1

