--- a/arch/arm/include/asm/elf.h	2010-03-08 10:45:44.000000000 -0800
+++ b/arch/arm/include/asm/elf.h	2010-03-30 15:26:47.000000000 -0700
@@ -98,6 +98,7 @@ extern int elf_check_arch(const struct e
 extern int arm_elf_read_implies_exec(const struct elf32_hdr *, int);
 #define elf_read_implies_exec(ex,stk) arm_elf_read_implies_exec(&(ex), stk)
 
+struct task_struct;
 int dump_task_regs(struct task_struct *t, elf_gregset_t *elfregs);
 #define ELF_CORE_COPY_TASK_REGS dump_task_regs
 
