From e07c5245ec8d313ea5352556a545c8c284224791 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 23 Mar 2012 10:20:08 -0500
Subject: [PATCH] mm: cma: out_of_memory now has a force_kill parameter

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 mm/page_alloc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index d344fef..bff16a6 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -5687,7 +5687,7 @@ static int __reclaim_pages(struct zone *zone, gfp_t gfp_mask, int count)
 						      NULL);
 		if (!did_some_progress) {
 			/* Exhausted what can be done so it's blamo time */
-			out_of_memory(zonelist, gfp_mask, order, NULL);
+			out_of_memory(zonelist, gfp_mask, order, NULL, false);
 		}
 	}
 
-- 
1.7.9.1

