From 59744f97e8b97ab06697b3f2bfa22f855c07034e Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 24 Feb 2012 11:27:27 -0600
Subject: [PATCH 1/2] cma-23 mm: compaction: export some of the functions

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 mm/internal.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/mm/internal.h b/mm/internal.h
index 55e7eed..aee4761 100644
--- a/mm/internal.h
+++ b/mm/internal.h
@@ -121,7 +121,7 @@ struct compact_control {
 	unsigned long migrate_pfn;	/* isolate_migratepages search base */
 	bool sync;			/* Synchronous migration */
 
-	unsigned int order;		/* order a direct compactor needs */
+	int order;			/* order a direct compactor needs */
 	int migratetype;		/* MOVABLE, RECLAIMABLE etc */
 	struct zone *zone;
 };
-- 
1.7.9

