From e46c14cc0f96a018137646a76bae36efa4bc7f76 Mon Sep 17 00:00:00 2001
From: Steve Sakoman <steve@sakoman.com>
Date: Mon, 10 May 2010 20:44:09 -0700
Subject: [PATCH 28/31] OMAP: DSS2: enable hsclk in dsi_pll_init for OMAP36XX

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/video/omap2/dss/dpi.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/drivers/video/omap2/dss/dpi.c b/drivers/video/omap2/dss/dpi.c
index 2d3ca4c..8703ea6 100644
--- a/drivers/video/omap2/dss/dpi.c
+++ b/drivers/video/omap2/dss/dpi.c
@@ -181,7 +181,12 @@ int omapdss_dpi_display_enable(struct omap_dss_device *dssdev)
 
 #ifdef CONFIG_OMAP2_DSS_USE_DSI_PLL
 	dss_clk_enable(DSS_CLK_SYSCK);
-	r = dsi_pll_init(dssdev, 0, 1);
+
+	if (cpu_is_omap3630())
+		r = dsi_pll_init(dssdev, 1, 1);
+	else
+		r = dsi_pll_init(dssdev, 0, 1);
+	
 	if (r)
 		goto err3;
 #endif
-- 
1.7.1

