From a10c1be29d7507d811033c996bda324b3d420427 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 23 Jan 2012 15:39:40 -0600
Subject: [PATCH] copy/paste error

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 mm/compaction.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/mm/compaction.c b/mm/compaction.c
index 278c069..6e4333a 100644
--- a/mm/compaction.c
+++ b/mm/compaction.c
@@ -1,4 +1,4 @@
-nr_isolated++;/*
+/*
  * linux/mm/compaction.c
  *
  * Memory compaction for the reduction of external fragmentation. Note that
-- 
1.7.4.1

