diff --git a/drivers/net/ks8851.c b/drivers/net/ks8851.c
index 1901a3d..5b060c2 100644
--- a/drivers/net/ks8851.c
+++ b/drivers/net/ks8851.c
@@ -563,7 +563,7 @@ static void ks8851_rx_pkts(struct ks8851_net *ks)
 			       ks->rc_rxqcr | RXQCR_SDA | RXQCR_ADRFE);
 
 		if (rxlen > 0) {
-			skb = netdev_alloc_skb(ks->netdev, rxlen + 2 + 8);
+			skb = netdev_alloc_skb(ks->netdev, rxlen + 4 + 8);
 			if (!skb) {
 				/* todo - dump frame and move on */
 			}
