From 00a6954707bc4d2ebc2a54770763f7ce59476376 Mon Sep 17 00:00:00 2001
From: Steve Sakoman <steve@sakoman.com>
Date: Mon, 4 Jan 2010 11:20:25 -0800
Subject: [PATCH 04/31] mmc: don't display single block read console messages

---
 drivers/mmc/card/block.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index cb9fbc8..c044648 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -346,8 +346,8 @@ static int mmc_blk_issue_rq(struct mmc_queue *mq, struct request *req)
 		if (brq.cmd.error || brq.data.error || brq.stop.error) {
 			if (brq.data.blocks > 1 && rq_data_dir(req) == READ) {
 				/* Redo read one sector at a time */
-				printk(KERN_WARNING "%s: retrying using single "
-				       "block read\n", req->rq_disk->disk_name);
+				/* printk(KERN_WARNING "%s: retrying using single "
+				       "block read\n", req->rq_disk->disk_name); */
 				disable_multi = 1;
 				continue;
 			}
-- 
1.7.1

