From 5b720444551455cf220b9ac204c3a985bead13e8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 30 Dec 2014 09:35:53 -0600
Subject: [PATCH] auto generated: capes: add dtbs to makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 3773f6f..9f00a7d 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -352,6 +352,21 @@ dtb-$(CONFIG_ARCH_OMAP3) += am3517-craneboard.dtb \
 dtb-$(CONFIG_SOC_AM33XX) += am335x-base0033.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-ttyO5.dtb \
+	am335x-boneblack-ttyO4.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
+	am335x-boneblack-can1.dtb \
+	am335x-boneblack-can0.dtb \
+	am335x-boneblack-bbb-exp-c.dtb \
+	am335x-bone-ttyO5.dtb \
+	am335x-bone-ttyO4.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-cape-bone-argus.dtb \
+	am335x-bone-can1.dtb \
+	am335x-bone-can0.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-nano.dtb \
-- 
2.1.4

