From 9bc9dc1d39fcd3974cdf93a1e94378b7956b1172 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 4 Jun 2015 10:32:29 -0500
Subject: [PATCH] auto generated: capes: add dtbs to makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c97bf83..75350bb 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -415,6 +415,16 @@ dtb-$(CONFIG_SOC_AM33XX) += \
 	am335x-base0033.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-universal.dtb \
+	am335x-boneblack-wl1835mod.dtb \
+	am335x-boneblack-replicape.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
+	am335x-boneblack-can0.dtb \
+	am335x-boneblack-bbb-exp-r.dtb \
+	am335x-boneblack-bbb-exp-c.dtb \
+	am335x-bone-cape-bone-argus.dtb \
+	am335x-bone-can0.dtb \
+	am335x-arduino-tre.dtb \
 	am335x-boneblack-overlay.dtb \
 	am335x-bonegreen.dtb \
 	am335x-evm.dtb \
-- 
2.1.4

