From 59756a3d74c54c9704be8fce9f576f33303ca2a1 Mon Sep 17 00:00:00 2001
From: Elias Bakken <elias@iagent.no>
Date: Fri, 10 Apr 2015 12:08:16 +0200
Subject: [PATCH] auto generated: capes: add dtbs to makefile

Signed-off-by: Elias Bakken <elias@iagent.no>
---
 arch/arm/boot/dts/Makefile | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 976a2e3..4ebfa68 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -397,6 +397,22 @@ dtb-$(CONFIG_SOC_AM33XX) += \
 	am335x-base0033.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-replicape.dtb \
+	am335x-boneblack-ttyO5.dtb \
+	am335x-boneblack-ttyO4.dtb \
+	am335x-boneblack-ttyO2.dtb \
+	am335x-boneblack-ttyO1.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
+	am335x-boneblack-can1.dtb \
+	am335x-boneblack-can0.dtb \
+	am335x-boneblack-bbb-exp-c.dtb \
+	am335x-bone-ttyO5.dtb \
+	am335x-bone-ttyO4.dtb \
+	am335x-bone-ttyO2.dtb \
+	am335x-bone-ttyO1.dtb \
+	am335x-bone-cape-bone-argus.dtb \
+	am335x-bone-can1.dtb \
+	am335x-bone-can0.dtb \
 	am335x-evm.dtb \
 	am335x-evmsk.dtb \
 	am335x-nano.dtb \
-- 
2.1.0

