From 7657001e8ccf57d5ece9e630b9bf1527374d9012 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 21 May 2015 10:58:41 -0500
Subject: [PATCH] auto generated: capes: add dtbs to makefile

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index e5481be..d9f02fa 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -419,6 +419,13 @@ dtb-$(CONFIG_SOC_AM33XX) += \
 	am335x-base0033.dtb \
 	am335x-bone.dtb \
 	am335x-boneblack.dtb \
+	am335x-boneblack-wl1835mod.dtb \
+	am335x-boneblack-replicape.dtb \
+	am335x-boneblack-cape-bone-argus.dtb \
+	am335x-boneblack-can0.dtb \
+	am335x-boneblack-bbb-exp-c.dtb \
+	am335x-bone-cape-bone-argus.dtb \
+	am335x-bone-can0.dtb \
 	am335x-boneblack-overlay.dtb \
 	am335x-bonegreen.dtb \
 	am335x-evm.dtb \
-- 
2.1.4

