From 3cbf130a3a6cd479ae00e6996d971402e22106f8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 26 Jan 2016 10:39:31 -0600
Subject: [PATCH 4/5] drm/adihdmi: Drop dummy save/restore hooks

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/gpu/drm/i2c/adihdmi_drv.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/drivers/gpu/drm/i2c/adihdmi_drv.c b/drivers/gpu/drm/i2c/adihdmi_drv.c
index 907bdf9..dd25f7d 100644
--- a/drivers/gpu/drm/i2c/adihdmi_drv.c
+++ b/drivers/gpu/drm/i2c/adihdmi_drv.c
@@ -956,16 +956,6 @@ static void adihdmi_encoder_mode_set(struct drm_encoder *encoder,
 			adihdmi->f_tmds = mode->clock;
 }
 
-static void adihdmi_encoder_restore(struct drm_encoder *encoder)
-{
-	pr_debug("%s - %d\n", __FUNCTION__, __LINE__);
-}
-
-static void adihdmi_encoder_save(struct drm_encoder *encoder)
-{
-	pr_debug("%s - %d\n", __FUNCTION__, __LINE__);
-}
-
 static void adihdmi_encoder_prepare(struct drm_encoder *encoder)
 {
 	adihdmi_encoder_dpms(encoder, DRM_MODE_DPMS_OFF);
@@ -978,8 +968,6 @@ static void adihdmi_encoder_commit(struct drm_encoder *encoder)
 
 static struct drm_encoder_helper_funcs adihdmi_encoder_helper_funcs = {
 	.dpms		= adihdmi_encoder_dpms,
-	.save		= adihdmi_encoder_save,
-	.restore	= adihdmi_encoder_restore,
 	.mode_fixup	= adihdmi_encoder_mode_fixup,
 	.prepare	= adihdmi_encoder_prepare,
 	.commit		= adihdmi_encoder_commit,
@@ -1049,8 +1037,6 @@ static struct drm_encoder_slave_funcs adihdmi_encoder_slave_funcs = {
 	.mode_fixup         = adihdmi_encoder_mode_fixup,
 	.mode_set           = adihdmi_encoder_mode_set,
 	.mode_valid         = adihdmi_encoder_mode_valid,
-	.restore            = adihdmi_encoder_restore,
-	.save               = adihdmi_encoder_save,
 	.set_config         = adihdmi_encoder_slave_set_config,
 	.set_property       = adihdmi_encoder_set_property,
 };
-- 
2.8.0.rc3

