From f6d8de96b48881da9766ee7b34fb7e69c420d3c2 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 26 Jan 2016 10:44:46 -0600
Subject: [PATCH 5/5] drm/adihdmi: Pass 'name' to drm_encoder_init()

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/gpu/drm/i2c/adihdmi_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i2c/adihdmi_drv.c b/drivers/gpu/drm/i2c/adihdmi_drv.c
index dd25f7d..3d38626 100644
--- a/drivers/gpu/drm/i2c/adihdmi_drv.c
+++ b/drivers/gpu/drm/i2c/adihdmi_drv.c
@@ -1133,7 +1133,7 @@ static int adihdmi_bind(struct device *dev, struct device *master, void *data)
 
 	drm_encoder_helper_add(&priv->encoder, &adihdmi_encoder_helper_funcs);
 	ret = drm_encoder_init(drm, &priv->encoder, &adihdmi_encoder_funcs,
-			DRM_MODE_ENCODER_TMDS);
+			DRM_MODE_ENCODER_TMDS, NULL);
 	if (ret)
 		goto err_encoder;
 
-- 
2.8.0.rc3

