From 9925712e7c37b5e557dd01d8a30446d777d84d60 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 28 Mar 2016 09:16:04 -0500
Subject: [PATCH 6/6] adihdmi_drv: reg_default -> reg_sequence

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/gpu/drm/i2c/adihdmi_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i2c/adihdmi_drv.c b/drivers/gpu/drm/i2c/adihdmi_drv.c
index 3d38626..6792224 100644
--- a/drivers/gpu/drm/i2c/adihdmi_drv.c
+++ b/drivers/gpu/drm/i2c/adihdmi_drv.c
@@ -60,7 +60,7 @@ struct adihdmi2 {
 };
 
 /* ADI recommended values for proper operation. */
-static const struct reg_default adihdmi_fixed_registers[] = {
+static const struct reg_sequence adihdmi_fixed_registers[] = {
 	{ 0x98, 0x03 },
 	{ 0x9a, 0xe0 },
 	{ 0x9c, 0x30 },
-- 
2.8.0.rc3

