From e3ad5ed90fdeb33992b32b38a4a6d2b2f97f54de Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 11 Sep 2012 17:02:34 -0500
Subject: [PATCH] beagle_xm: cpufreq debug

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/plat-omap/clock.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/plat-omap/clock.c b/arch/arm/plat-omap/clock.c
index 20b2de8..e3a9ef8 100644
--- a/arch/arm/plat-omap/clock.c
+++ b/arch/arm/plat-omap/clock.c
@@ -128,11 +128,13 @@ int clk_set_rate(struct clk *clk, unsigned long rate)
 	if (!arch_clock || !arch_clock->clk_set_rate)
 		return ret;
 
+	pr_info("cpufreq: clk_set_rate: requesting: %ld\n", rate);
 	spin_lock_irqsave(&clockfw_lock, flags);
 	ret = arch_clock->clk_set_rate(clk, rate);
 	if (ret == 0)
 		propagate_rate(clk);
 	spin_unlock_irqrestore(&clockfw_lock, flags);
+	pr_info("cpufreq: clk_set_rate: setting: %ld Success\n", rate);
 
 	return ret;
 }
-- 
1.7.10.4

